### PR TITLE
Automating the Mintlify OpenAPI docs

### LINKED ISSUE #13907: Automate the production of the Mintlify OpenAPI docs
We've got a manual process now that involves:

1. Producing the `schema.json` file using our script and FastAPI's tooling
2. Adding a few fields to the file to make it amenable to Mintlify's scraper
3. Running the `mintlify-scrape openapi-file ...` command to produce the markdown docs
4. Taking the produced markdown docs and merging them with the global `mint.json`

Let's get this automated so that it can be done with a single command, and ultimately, be done in CI

### PR DESCRIPTION
**Pull request by chrisguidry**
This script will reconcile the current state of the REST API with what's in
the `docs/<version>/api-ref/server` directory, as well as the navigation within
the `mint.json` file that defines the overall documentation.

I'm also wiring this script up as a pre-commit hook so that any changes to the
API and schemas are automatically reflected in the docs.  To make this the least
invasive as I can, this attempts to install the required `@mintlify/scraping`
package if it is not currently installed.

Fixes #13907

### CODE REVIEW

File: .github/workflows/static-analysis.yaml [hunk 1]
```diff
@@ -30,12 +30,33 @@ jobs:
...
+          uv pip install --upgrade --system -e .[dev]
+
+      - name: Setup NodeJS
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: ".nvmrc"
+          cache-dependency-path: "**/package-lock.json"
+
+      - name: Install mintlify-scrape
+        run: npm install -g @mintlify/scraping
```
Comment (discdiver) [inline .github/workflows/static-analysis.yaml:59]: Do you need to install mintlify too? Not sure if it's now in our required development packages.

File: scripts/generate_mintlify_openapi_docs.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+PageGroup = dict[str, Any]
+
+
+def docs_path() -> Path:
+    return Path(__file__).parent.parent / "docs"
+
+
+def current_version() -> str:
+    # TODO: derive this from the docs and the current version of the prefect package
+    return "3.0rc"
```
Comment (discdiver) [inline scripts/generate_mintlify_openapi_docs.py:22]: Gotcha.

**Review by discdiver — APPROVED** [review]
Thank you! LGTM - other than failing pre-commit.  If not planning to address automatic version number setting as part of this work, we should probably open an issue for it.

File: .github/workflows/static-analysis.yaml (hunk 1 above)
Comment (chrisguidry) [reply .github/workflows/static-analysis.yaml:59]: For this particular script, we only need `mintlify-scrape`

File: scripts/generate_mintlify_openapi_docs.py (hunk 2 above)
Comment (chrisguidry) [reply scripts/generate_mintlify_openapi_docs.py:22]: Yeah this is something I want to dig more into.  I can make it work on `<major>.<minor>` for now (adding the `rc` tag if appropriate)

**Pull request merged**